src/lib/compat/cryptojs.ts

```diff
@@ -0,0 +1,148 @@
+/**
+ * This is a compatibility layer for decrypting AES encrypted base64 encoded
+ * strings from CryptoJS.AES.encrypt
+ *
+ * based on crypto-js 3.1.9-1
+ *
+ * The wallet used to depend on crypto-js for en/decryption of the seed, but
+ * then migrated to using the jolocom-lib (which uses crypto-browserify). The
+ * details of the encryption (IV, key derivation) are slightly different, so
+ * previously stored encrypted seeds needed to be re-encrypted.
+ *
+ */
+
+const CryptoJS = {
+  AES: {
+    decrypt: AESDecrypt,
+  },
+  algo: {
+    AES: {
+      keySize: 256/32,
+      ivSize: 128/32
+    }
+  },
+  kdf: {
+    OpenSSL: {
+      execute: OpenSSLKdfExecute
+    }
+  },
+  format: {
+    OpenSSL: {
+      parse: OpenSSLFormatterParse
+    }
+  }
+}
+
+export default CryptoJS
+
+import { randomBytes, createDecipheriv, createHash } from 'crypto'
+import { JolocomLib } from 'jolocom-lib';
+
+/**
+ * This function emulates CryptoJS.format.OpenSSL.parse (cipher-core.js)
+ * It parses a base64 cipher text and separates the salt bytes if any
+ */
+function OpenSSLFormatterParse(cipherTextBase64: string) {
+  let salt, cipherText = Buffer.from(cipherTextBase64, 'base64')
+  // Test for salt
+  if (cipherText.readUInt32BE(0*4) == 0x53616c74 && cipherText.readUInt32BE(1*4) == 0x65645f5f) {
+    // Extract salt, 64 bytes, word  2 and word 3
+    // NOTE: slice is byte addressed
+    salt = cipherText.slice(2 * 4, 4 * 4)
+    // Remove salt from ciphertext
+    cipherText = cipherText.slice(4 * 4)
+  }
+
+  // NOTE: ciphertext small 't' sic from CryptoJS
+  return { salt, ciphertext: cipherText }
+}
+
+/**
+ * This function emulates CryptoJS.kdf.OpenSSL.execute (cipher-core.js)
+ * It extracts a key and IV for the AES algorithm
+ */
+function OpenSSLKdfExecute(password: string, keySize: number, ivSize: number, salt: Buffer | undefined) {
+  // Generate random salt
+  if (!salt) {
+    salt = randomBytes(64/8)
+  }
+
+  // Derive key and IV
+  var key = EvpKDFCompute(password, salt, { keySize: keySize + ivSize, iterations: 1 });
+
+  // Separate key and IV
+  var iv = key.slice(keySize * 4)
+  key = key.slice(0, keySize * 4)
+
+  return { key, iv }
+}
+
+/**
+ * This function emulates CryptoJS.algo.EvpKDF.compute
+ */
+function EvpKDFCompute(password: string, salt: Buffer, cfg: { keySize: number, iterations: number }) {
+  // Init hasher
+  const newHasher = () => createHash('md5')
+
+  var keySize = cfg.keySize;
+  var iterations = cfg.iterations;
+
+  // Initial values
+  let derivedKey, block
+
+  // Generate key
+  for (let k = 0; k < keySize; k++) {
+    const hasher = newHasher()
+    if (block) hasher.update(block)
+    // @ts-ignore
+    block = hasher.update(password).update(salt).digest('buffer')
+
+    // Iterations
+    for (var i = 1; i < iterations; i++) {
```

Comment:
Does this ever run given the `iterations` is set to 1? Could be removed I suppose (if we want to keep it to make it easier to compare this to the `CryptoJS` code, we could add a note that this does not run), unless of course this runs somewhere and I missed it.

Reply:
`iterations` is 1 by default and we never change it, so yeah this could probably be commented out, will do that.